cvm/vllm_proxy/proxy_app.py

```diff
@@ -0,0 +1,182 @@
+import os
+import time
+import httpx
+import logging
+
+from fastapi import FastAPI, Request
+from fastapi.responses import Response, JSONResponse
+
+from utils import *
+
+# Configuration
+MACHINE = os.getenv("MACHINE", "H100")
+IS_CVM = os.getenv("IS_CVM", "false")
+
+# Model Configuration
+MODEL_ID = os.getenv("MODEL_ID", "openai/gpt-oss-120b")
+URL_BASE_VLLM = os.getenv("BASE_VLLM_URL", "http://vllm:8000")
+DEFAULT_MAX_TOKENS = int(os.getenv("MAX_TOKENS", "256"))
+DEFAULT_TEMPERATURE = float(os.getenv("TEMPERATURE", "0.2"))
+
+# Benchmarking
+BENCHMARK_FILE = "benchmark_backend.csv"
+
+# TEE Prompts
+TEE_DOC = load_txt("./tee_info.txt")
+TEE_INSTRUCTION = load_txt("tee_instruction_prompt.txt")
+BASE_SYSTEM_CORE = load_txt("base_system_prompt.txt")
+IS_TEE_RELATED_PROMPT = load_txt("is_tee_related_prompt.txt")
+
+# --------- Logging ----------
+logging.basicConfig(
+    level=os.getenv("LOG_LEVEL", "INFO"),
+    format="%(asctime)s %(levelname)s %(message)s",
+)
+logger = logging.getLogger("vLLM Gateway")
+
+# ---------- App ----------
+app = FastAPI(title="vLLM Gateway")
+
+# ---------------------- BenchmarkLogger ----------------------
+benchmark_logger = BenchmarkLogger(
+    file_path=Path(BENCHMARK_FILE),
+    columns=[
+        "is_cvm",
+        "machine",
+        "user_id",
+        "date",
+        "total_call_time_s",
+        "call1_time_s",
+        "call2_time_s",
+        "model",
+        "model_arguments",
+        "nb_query",
+    ],
+    logger=logger,
+    reset=False,
+)
+
+
+@app.get("/")
+async def root():
+    return {"Proxy api for vLLM"}
+
+
+@app.get("/health")
+def health_check():
+    return {"status": "ok"}
+
+
+@app.get("/v1/models")
+async def models():
+    t0 = time.perf_counter()
+    async with httpx.AsyncClient(timeout=httpx.Timeout(10.0, read=120.0)) as c:
+        r = await c.get(f"{URL_BASE_VLLM}/v1/models")
+    dt = time.perf_counter() - t0
+    logger.info(f"[Endpoint | /v1/models]: {r.status_code} in {dt:.1f}ms")
+    return Response(r.content, status_code=r.status_code,
+                    media_type=r.headers.get("content-type"))
+
+
+@app.post("/v1/chat/completions")
+async def chat(req: Request):
+    # expected : {"prompt": "...", "document": "...", "user_id": "..."}
+    dt_call2 = None
+    nb_query = 1
+    global_t0 = time.perf_counter()
+
+    try:
+        data = await req.json()
+    except Exception as e:
+        logger.error("❌ Invalid JSON: %s", e)
+        return JSONResponse({"error": "invalid JSON"}, status_code=400)
+
+    assert 'prompt' in data, '❌ Bad request keys'
+    assert 'document' in data, '❌ Bad request keys'
+    assert 'user_id' in data, '❌ Bad request keys'
```

Comment:
I thought I commented that already. You can use fastapi capabilities to do such assertion. It will also return a proper error to the client.